staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -311,13 +311,16 @@ func (req *request) NoteQueued(inQueue bool) {
 	}
 }
 
-func (req *request) Finish(execFn func()) bool {
+func (req *request) Finish(execFn func()) (idle bool) {
 	exec, idle := req.wait()
 	if !exec {
-		return idle
+		return
 	}
+	defer func() {
+		idle = req.qs.finishRequestAndDispatchAsMuchAsPossible(req)
```

Comment:
tweaking return value in a defer block looks tricky to me, how about:

```go
exec, _ := req.wait()
if !exec {
    return
}
defer req.qs.finishRequestAndDispatchAsMuchAsPossible(req)
execFn()
return req.qs.IsIdle()
```

Reply:
Modifying a result variable in a deferred func does not seem very tricky to me.  Rule 4 in https://golang.org/ref/spec#Block says

> The scope of an identifier denoting a method receiver, function parameter, or result variable is the function body.

The suggested replacement acquires the QueueSet's mutex twice, while the original does so only once.

Reply:
This looks tricky to me too.  In library-go in openshift it's inscrutable.  I'm not seeing the reason for this change either unless there is a separate concern of execFn panicing?  In which case wrap the panic risk.